nightfall-client/src/services/withdraw.mjs

```diff
@@ -28,7 +33,7 @@ const NEXT_N_PROPOSERS = 3;
 async function withdraw(withdrawParams) {
   logger.info('Creating a withdraw transaction');
   // let's extract the input items
-  const { offchain = false, ...items } = withdrawParams;
+  const { offchain = false, providedCommitments, ...items } = withdrawParams;
```

Comment:
I think that `providedCommitments` should be an array of commitment hashes. Therefore, in order to work properly with the current logic, you will need to get the whole commitment information from the database. Currently, there's no function that gets a commitment by hash, so you'll need to create one and add it to `commitment-storage.mjs`. You can take a look at how `getCommitmentBySalt` works. Then, you'll need to check that the user is the owner of that commitment and otherwise throw an error.

Reply:
this is something that I wanted to discuss with you as well. I think having an array of hashes instead of the full objects is a good idea. I basically already have that implemented in the test I'm writing so I'll work on that once the tests are done.

Reply:
Great! Passing `providedCommitments` as an array of hashes and handling the commitments inside the client code will simplify things on the user side :)